src/cmds/play.c

```diff
@@ -14,10 +14,17 @@
 #include <drivers/audio/portaudio.h>
 #include <fs/file_format.h>
 #include <util/math.h>
+#include <framework/mod/options.h>
+
+#define PLAY_FROM_FLASH OPTION_GET(BOOLEAN, play_from_flash)
```

Comment:
I think it's better to replace this option with something like `USE_LOCAL_BUFFER` and just fail it there's not local buffer, but user requests playing audio from file. This approach will reduce number of `#ifdef .. #else .. #endif`

Reply:
Thank you, I will try something like this

Reply:
We can implement two different functions - play_from_memory and play_from_file. What do you think? In this case we can get rid from this ifdef's. Because now we must maintain both buffers and files in the same function which leads to ifdef's...